### PR TITLE
Update SDK to match API definitions

### PR DESCRIPTION
**Pull request by mathewsonke**
This PR is a collection of small inconsistencies I found when transplanting the SDK examples. 

- Fixes a typo in the `Files` resource
- Removes the `Notes` and `Custom Events` resources
- Removes various methods which refer to endpoints that do not exist in the api-definitions
- Modifies a method name to use American English

When removing the methods, I took the naive approach of considering `api-definitions` as the Source of Truth here. Chances are at least one of those deletions is incorrect- if that's the case I can instead open a bug report for missing documentation in `api-definitions`

### CODE REVIEW

File: src/resources/coupons-resource.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export default function CouponsResource({apiHandler}) {
             return apiHandler.post(`coupons-redemptions`, data);
         },
 
-        setExpiry({redemptionCode, data}) {
+        setExpiration({redemptionCode, data}) {
```
Comment (cesarlevel) [inline src/resources/coupons-resource.js:49]: This means we need to make the changes on Recomm and tell merchants that are using this.

Comment (mathewsonke) [reply src/resources/coupons-resource.js:49]: Yeah, although it's insulated behind a major version change. There are several breaking changes in this PR; my thinking was we can do a major release which brings the JS-SDK to be fully consistent with `api-definitions`

But I'm very interested in getting feedback on that! This PR was easy to throw together while working on the SDK docs, so I consider it somewhat of a proposal

Comment (cesarlevel) [reply src/resources/coupons-resource.js:49]: Gotcha, thanks! Same here I would love to hear from others about this subject

Comment (mathewsonke) [reply src/resources/coupons-resource.js:49]: So looking through the API logs, it doesn't seem like this endpoint has ever been meaningfully been used on live or sandbox, it's almost entirely 4xx errors. It's not present in the Recomm codebase either. 

Maybe we can just drop it entirely?

Comment (mattsanf) [reply src/resources/coupons-resource.js:49]: I'd like to see us start to use a two-step approach to removing and refactoring endpoint names like this.

one major release to throw a depreciation notice, then after some time another major release to remove it.

the old function could look something like this

```
setExpiry() {
  console.warn('"setExpiry" function is deprecated and will be remove with the next major release version. Please use "setExpiration" instead. See docs www.docs.rebilly.com');
  setExpiration(arguments);
}
```

Comment (mathewsonke) [reply src/resources/coupons-resource.js:49]: For the majority of situations I completely agree with an intermediate deprecation step, but I don't believe anyone has ever actually used the `setExpiry` method anywhere. The endpoint itself isn't really used at all, let alone SDK method.

That's why I think a rename or outright removal in a single major version would be okay here.

Comment (ZengJunyong) [reply src/resources/coupons-resource.js:49]: Basically I agree with @mathewsonke , renaming it is easy if no one is using it

Comment (adamaltman) [reply src/resources/coupons-resource.js:49]: > This means we need to make the changes on Recomm and tell merchants that are using this.

@cesarlevel no. This means Recomm has to make changes only if it upgrades to the next major version of the js-sdk. There is a big nuance there, and it's really important to see it.

I actually want to remove the js-sdk entirely. More on that in an RFC.

**Pull request merged**